File: Rudder/RSLibraryInfo.m
//
//  RSLibraryInfo.m
//  RSSDKCore
//
//  Created by Arnab Pal on 17/10/19.
//  Copyright © 2019 RSlabs. All rights reserved.
//

#import "RSLibraryInfo.h"
#import "RSConstants.h"

@implementation RSLibraryInfo
- (instancetype)init
{
    self = [super init];
    if (self) {
        _name = @"rudder-ios-library";
        _version = RS_VERSION;
    }
    return self;
}

- (NSDictionary<NSString *,NSObject *> *)dict {
    NSMutableDictionary *tempDict = [[NSMutableDictionary alloc] init];
    
    [tempDict setValue:_name forKey:@"name"];
    [tempDict setValue:_version forKey:@"version"];
    
    return [tempDict copy];
}

@end

File: Rudder/Rudder/Domain/Models/RSLibraryInfo.swift
//
//  RSLibraryInfo.swift
//  Rudder
//
//  Created by Desu Sai Venkat on 06/08/21.
//  Copyright © 2021 Rudder Labs India Pvt Ltd. All rights reserved.
//

import Foundation

struct RSLibraryInfo{
    let name: String
    let version: String
    
    init() {
        name = "rudder_ios_library"
        version = Constants.RS_VERSION
    }
    
    func dict() -> [String:String]
    {
        var tempDict:[String:String] = [:]
        tempDict["name"] = name
        tempDict["version"] = version
        return tempDict
    }
}
